cookbook/lvgl.rst

```diff
@@ -0,0 +1,1655 @@
+                          no_repeat: true
+                      - text: 6
+                        control:
+                          no_repeat: true
+                  - buttons:
+                      - text: 7
+                        control:
+                          no_repeat: true
+                      - text: 8
+                        control:
+                          no_repeat: true
+                      - text: 9
+                        control:
+                          no_repeat: true
+                  - buttons:
+                      - text: "\uF55A"
+                        key_code: "*"
+                        control:
+                          no_repeat: true
+                      - text: 0
+                        control:
+                          no_repeat: true
+                      - text: "\uF00C"
+                        key_code: "#"
+                        control:
+                          no_repeat: true
+
+    key_collector:
+      - source_id: lvgl_keypad
+        min_length: 4
+        max_length: 4
+        end_keys: "#"
+        end_key_required: true
+        back_keys: "*"
+        allowed_keys: "0123456789*#"
+        timeout: 5s
+        on_progress:
+          - if:
+              condition:
+                lambda: return (0 != x.compare(std::string{""}));
+              then:
+                - lvgl.label.update:
+                    id: lvgl_label
+                    text: !lambda 'return x.c_str();'
+              else:
+                - lvgl.label.update:
+                    id: lvgl_label
+                    text: "Enter code and \uF00C"
+        on_result:
+          - if:
+              condition:
+                lambda: return (0 == x.compare(std::string{"1234"}));
+              then:
+                - lvgl.led.update:
+                    id: lvgl_led
+                    color: 0x00FF00
+              else:
+                - lvgl.led.update:
+                    id: lvgl_led
+                    color: 0xFF0000
+
+A few notable things in this example: usage of a base object ``obj`` as a parent for the label (in order to center the label in the middle of it and emphasize it with shadows independently of the label's dimensions); usage of ``align_to`` to align it to the led vertically; changing the background color of the buttons in ``pressed`` state; using the ``key_code`` configuration option to send a different character to ``key_collector`` instead of the displayed symbol.
+
+.. _lvgl-cook-idlescreen:
+
+Turn off screen when idle
+-------------------------
+
+LVGL has a notion of screen inactivity, i.e. how long did the user not interact with the screen. This can be used to dim the display backlight or turn it off after a moment of inactivity (like a screen saver). Touching the screen counts as an activity and resets the inactivity counter (it's important to use the ``on_release`` trigger). With a template number you can make the timeout settable by the users.
```

Comment:
```suggestion
LVGL has a notion of screen inactivity -- in other words, the time since the last user interaction with the screen is tracked. This can be used to dim the display backlight or turn it off after a moment of inactivity (like a screen saver). Every use of an input device (touchscreen, rotary encoder) counts as an activity and resets the inactivity counter. Note that it's important to use the ``on_release`` trigger to accomplish this task. With a template number you can make the timeout adjustable by the users.
```